internal/engine/image_build_and_deployer.go

```diff
@@ -273,16 +272,26 @@ func (ibd *ImageBuildAndDeployer) indentLogger(ctx context.Context) context.Cont
 	return logger.WithLogger(ctx, newL)
 }
 
+type k8sDeploySet struct {
+	entities              []k8s.K8sEntity
+	syncletInjected       bool
+	syncletInjectionError bool
+}
+
+func (s k8sDeploySet) HasEligibleSynclet() bool {
+	return s.syncletInjected && !s.syncletInjectionError
+}
+
 func (ibd *ImageBuildAndDeployer) createEntitiesToDeploy(ctx context.Context,
 	iTargetMap map[model.TargetID]model.ImageTarget, k8sTarget model.K8sTarget,
-	results store.BuildResultSet, needsSynclet bool) ([]k8s.K8sEntity, error) {
-	newK8sEntities := []k8s.K8sEntity{}
+	results store.BuildResultSet, needsSynclet bool) (k8sDeploySet, error) {
+	result := k8sDeploySet{}
 
 	// TODO(nick): The parsed YAML should probably be a part of the model?
 	// It doesn't make much sense to re-parse it and inject labels on every deploy.
 	entities, err := k8s.ParseYAMLFromString(k8sTarget.YAML)
 	if err != nil {
-		return nil, err
+		return result, err
```

Comment:
what's the point of returning the result along with these errors?

Reply:
fixed! now it returns nil